client/patch-react.js

```diff
@@ -93,3 +110,10 @@ function wrap (fn, around) {
 
   return _fn
 }
+
+function canOverrideRender (prototype) {
+  const descriptor = Object.getOwnPropertyDescriptor(prototype, 'render')
```

Comment:
You need to walk up the prototype chain for a proper check, in case the component has been sub-classed, i.e. `render()` is defined on a super class (`getOwnPropertyDescriptor` only retrieves the descriptor for the immediate object). I'd also recommend to patch on exactly that prototype object.